src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx

```diff
@@ -1,21 +1,188 @@
+import AppAutoComplete from '~/components/app-auto-complete/AppAutoComplete'
+import axios from 'axios'
+import { useStepContext } from '~/context/step-context'
+
+const COUNTRIES_API_URL = 'https://restcountries.com/v3.1/all'
+const CITIES_API_URL = 'http://api.geonames.org/searchJSON'
 
 const GeneralInfoStep = ({ btnsBox }) => {
   const { t } = useTranslation()
+  const { stepData, handleGeneralInfo } = useStepContext()
+  const generalInfo = stepData.generalInfo
+
+  const [countries, setCountries] = useState([])
+  const [countryCodes, setCountryCodes] = useState({})
+  const [cities, setCities] = useState([])
+  const [data, setData] = useState({
+    firstName: generalInfo.data.firstName || '',
+    lastName: generalInfo.data.lastName || '',
+    country: generalInfo.data.country || '',
+    city: generalInfo.data.city || '',
+    professionalSummary: generalInfo.data.professionalSummary || ''
+  })
+
+  useEffect(() => {
+    axios
+      .get(COUNTRIES_API_URL)
+      .then((response) => {
+        const countryNames = response.data
+          .map((country) => {
+            setCountryCodes((prevCodes) => ({
+              ...prevCodes,
+              [country.name.common]: country.cca2
+            }))
+            return country.name.common
+          })
+          .sort((a, b) => a.localeCompare(b))
+        setCountries(countryNames)
+      })
+      .catch((error) => {
+        console.error('Error fetching countries:', error)
+      })
+  }, [])
+
+  useEffect(() => {
+    const fetchCities = (country) => {
+      const countryCode = countryCodes[country]
+
+      if (countryCode) {
+        axios
+          .get(
+            `${CITIES_API_URL}?formatted=true&country=${countryCode}&maxRows=10&username=geer_sann`
+          )
+          .then((response) => {
+            console.log(response.data)
+            const cityNames = response.data.geonames.map((city) => city.name)
+            setCities(cityNames)
+          })
+          .catch((error) => {
+            console.error('Error fetching cities:', error)
+          })
+      } else {
+        setCities([])
+      }
+    }
+
+    if (data.country) {
+      fetchCities(data.country)
+    } else {
+      setCities([])
+    }
+  }, [data.country, countryCodes])
+
+  useEffect(() => {
+    handleGeneralInfo({ data })
+  }, [data, handleGeneralInfo])
+
+  const handleInputChange = (field) => (event) => {
+    setData({ ...data, [field]: event.target.value })
+  }
+
+  const filterOptions = (options, state) => {
+    const defaultFilterOptions = createFilterOptions()
+    return defaultFilterOptions(options, state).slice(0, 300)
+  }
+
   return (
     <Box sx={styles.container}>
-      <Box component='form' sx={styles.form}>
-        <Box>
+      <Grid alignItems='center' container justifyContent='center' spacing={4}>
+        <Grid item md={6} sx={{ display: { xs: 'none', md: 'block' } }} xs={12}>
+          <Box sx={styles.imgContainer}>
```

Comment:
FYI, there’s a `StepImage` component available at `/Client/src/components/step-image/StepImage.tsx` – you can use it for code consistency if you wish

Reply:
Probably it's make sense. Updated it